src/dhcp/clientv4.rs

```diff
@@ -114,6 +117,25 @@ impl Client {
         self.next_egress - now
     }
 
+    /// Check if the existing IP address lease has expired.
+    ///
+    /// # Note
+    /// RC 2131 requires that a client immediately cease usage of an
+    /// address once the lease has expired.
+    ///
+    /// # Args
+    /// * `now` - The current network time instant.
+    ///
+    /// # Returns
+    /// True if a lease exists and it has expired. False otherwise.
+    pub fn lease_expired(&self, now: Instant) -> bool {
```

Comment:
Instead of having a manual API to check, we could have the client manually return a config with an UNSPECIFIED Ipv4 address so that the config is stuck. This is similar to what happens if the DHCP Retry mechanism is exceeded.

Reply:
IMO returning UNSPECIFIED is better than a manual `lease_expired` API. I'd say most users care mostly about whether DHCP is up or not (do we have a valid lease or not), and care less about the reason. Also, it's less work for the user.